silhouette/app/com/mohiva/play/silhouette/impl/providers/oauth2/GitLabProvider.scala

```diff
@@ -0,0 +1,155 @@
+/**
+ * Original work: SecureSocial (https://github.com/jaliss/securesocial)
+ * Copyright 2013 Jorge Aliss (jaliss at gmail dot com) - twitter: @jaliss
+ *
+ * Derivative work: Silhouette (https://github.com/mohiva/play-silhouette)
+ * Modifications Copyright 2015 Mohiva Organisation (license at mohiva dot com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mohiva.play.silhouette.impl.providers.oauth2
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.api.util.HTTPLayer
+import com.mohiva.play.silhouette.impl.exceptions.ProfileRetrievalException
+import com.mohiva.play.silhouette.impl.providers._
+import com.mohiva.play.silhouette.impl.providers.oauth2.GitLabProvider._
+import play.api.http.HeaderNames
+import play.api.libs.json.JsValue
+
+import scala.concurrent.Future
+
+/**
+ * Base GitLab OAuth2 Provider.
+ *
+ * @see https://gitlab.com/help/api/oauth2.md
+ *      https://gitlab.com/help/integration/oauth_provider.md
+ */
+trait BaseGitLabProvider extends OAuth2Provider {
+
+  /**
+   * The content type to parse a profile from.
+   */
+  override type Content = JsValue
+
+  /**
+   * The provider ID.
+   */
+  override val id = ID
+
+  /**
+   * Defines the URLs that are needed to retrieve the profile data.
+   */
+  override protected val urls = Map("api" -> settings.apiURL.getOrElse(API))
+
+  /**
+   * A list with headers to send to the API.
+   *
+   * Without defining the accept header, the response will take the following form:
+   * access_token=e72e16c7e42f292c6912e7710c838347ae178b4a&scope=user%2Cgist&token_type=bearer
+   *
+   * @see https://developer.github.com/v3/oauth/#response
+   */
+  override protected val headers = Seq(HeaderNames.ACCEPT -> "application/json")
```

Comment:
It is Github related, I have to admit that I haven't tried without it, I'm gonna try it now.
I also noticed that I forgot to change the ID and Callback API, will do it now.